lib/mastermind_new.rb

```diff
@@ -0,0 +1,165 @@
+module Mastermind
+    def self.play
+        previous_guesses = []
+        remaining_guesses = 10
+        pins_out = []
+        gen_list = GeneratedList.new
+        game_code = Code.new(gen_list.random_code)
+        until game_code.matched do
+            system "clear"
+            if remaining_guesses == 0
+                puts "You ran out of guesses..."
+                break
+            end
+            if previous_guesses != []
+                puts "Previous Guesses:"
+                previous_guesses.map do |v|
+                    puts v.join(' ')
+                end
+            end
+            if pins_out != []
+                puts "Red Pins: " + pins_out[0].to_s
+                puts "White Pins: " + pins_out[1].to_s
+            end
+            puts remaining_guesses.to_s + " Guesses Remaining..."
+            puts "Your color choices are R, G, Y, B, M, and C."
+            user_in = Input.new
+            until user_in.is_valid do
+                user_in = Input.new
+            end
+            current_code = Code.new(user_in.user_input)
+            previous_guesses << current_code.convert
+            pins_out = game_code.compare(current_code.value)
+            remaining_guesses -= 1
+        end
+        puts "The code was: " + game_code.convert.join(' ')
+    end
+
+    class GeneratedList
+        attr_reader :get
```

Comment:
Might consider renaming @get to something like @all_possible_secret_codes? Not clear what this variable means without reading the GeneratedList#initialize method.